internal/middleware/requestid.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/openfga/openfga/pkg/logger"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const requestIDCtxKey ctxKey = "request-id-context-key"
+
+func NewRequestIDInterceptor(logger logger.Logger) grpc.UnaryServerInterceptor {
```

Comment:
Mind adding a simple test for this middleware?

Reply:
Now that was interesting. First time writing a middleware test.